diff --git a/Python/ceval_gil.h b/Python/ceval_gil.h
index ef5189068e..da497a1679 100644
--- a/Python/ceval_gil.h
+++ b/Python/ceval_gil.h
@@ -5,6 +5,7 @@
 #include <stdlib.h>
 #include <errno.h>
 
+#include "/usr/local/cuda/include/nvtx3/nvToolsExt.h"
 
 /* First some general settings */
 
@@ -148,6 +149,7 @@ static void recreate_gil(void)
 
 static void drop_gil(PyThreadState *tstate)
 {
+    nvtxRangePushA("drop_gil");
     if (!_Py_atomic_load_relaxed(&_PyRuntime.ceval.gil.locked))
         Py_FatalError("drop_gil: GIL is not locked");
     /* tstate is allowed to be NULL (early interpreter init) */
@@ -246,6 +248,7 @@ _ready:
 
     MUTEX_UNLOCK(_PyRuntime.ceval.gil.mutex);
     errno = err;
+    nvtxRangePop();
 }
 
 void _PyEval_SetSwitchInterval(unsigned long microseconds)
